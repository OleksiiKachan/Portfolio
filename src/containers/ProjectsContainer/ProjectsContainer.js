import React, { Component } from 'react';
import ProjectsSection from '../../components/ProjectsSection/ProjectsSection.js'

/*
    <ProjectsContainer projects={} />
*/
class ProjectsContainer extends Component {
    render() {
        return (
            <ProjectsSection projects={this.props.projects}/>
        );
    }
}

export default ProjectsContainer;