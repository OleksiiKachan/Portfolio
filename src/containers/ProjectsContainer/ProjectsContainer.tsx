import React, { Component } from 'react';
import ProjectsSection from '../../components/ProjectsSection';

/*
    <ProjectsContainer projects={} />
*/
class ProjectsContainer extends Component<any, any> {
  render() {
    const dataStore = require('../../data.json');
    return <ProjectsSection projects={dataStore.projects} />;
  }
}

export default ProjectsContainer;
