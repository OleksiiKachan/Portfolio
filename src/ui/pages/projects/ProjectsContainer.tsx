import React, { Component } from 'react';
import ProjectsSection from './ProjectsSection/ProjectsSection';

/*
    <ProjectsContainer />
*/

export default class ProjectsContainer extends Component<{}> {
  render() {
    const dataStore = require('../../../data.json');
    return <ProjectsSection projects={dataStore.projects} />;
  }
}
