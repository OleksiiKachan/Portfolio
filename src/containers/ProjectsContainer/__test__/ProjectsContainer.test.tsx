import React from 'react';
import { shallow } from 'enzyme';
import ProjectsContainer from '..';

describe('ProjectsContainer', () => {
  it('Snapshot', () => {
    const component = shallow(<ProjectsContainer />);
    expect(component).toMatchSnapshot();
  });
});
