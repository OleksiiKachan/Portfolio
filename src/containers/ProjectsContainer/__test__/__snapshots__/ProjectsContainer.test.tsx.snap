// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectsContainer Snapshot 1`] = `
<ProjectsSection
  projects={
    Array [
      Object {
        "client": Array [
          Object {
            "name": "Hyundai Motor Manufacturing Alabama",
            "url": "https://www.hmmausa.com/",
          },
        ],
        "coverImage": "https://res.cloudinary.com/oleksiikachan/image/upload/v1557516569/portfolio/images/projects/agw/card.png",
        "externalResources": Array [
          Object {
            "name": "Product Page. AGW MES",
            "url": "http://olsom.net/agw-mes/",
          },
          Object {
            "name": "Product Page. AGW WMS",
            "url": "http://olsom.net/agw-wms/",
          },
          Object {
            "name": "Product Page. AGW IP",
            "url": "http://olsom.net/agw-ip/",
          },
          Object {
            "name": "Company Page. OLSOM LLC",
            "url": "http://olsom.net/",
          },
        ],
        "id": "agw",
        "images": Array [
          Object {
            "imageUrl": "https://res.cloudinary.com/oleksiikachan/image/upload/v1557517324/portfolio/images/projects/agw/cover.png",
          },
        ],
        "longStack": Array [
          ".NET Framework",
          "C#",
          "ASP.NET MVC",
          "ASP.NET Web Forms",
          "Javascript",
          "jQuery",
          "MS SQL Server",
          "Oracle",
          "Windows Service",
        ],
        "partner": Array [
          Object {
            "name": "OLSOM LLC",
            "url": "http://olsom.net/",
          },
        ],
        "projectDescription": Array [
          Object {
            "content": "|italic|AGW|-| software addresses the challenges of |bold|workflow automation, real-time data collection, system integration|-| and business insight. It offers scalable, high-performance, reliable and |bold|flexible solutions|-| that meet the demand of intelligent process automation for manufacturers, 3PLs, distributors and other spheres of business.",
            "type": "string",
          },
          Object {
            "content": "Ranging a wide set of solutions from a small tailored operator support systems or system integration platforms to powerful business decision-making support environments AGW offers scalable execution network for enterprise.",
            "type": "string",
          },
          Object {
            "alignment": "left",
            "content": "|italic|AGW IP|-| is a complex manufacturing environment that deals with significant volumes of heterogeneous design and production data. Efficient collection, consolidation, transformation and delivery of these data in secure and timely manner is a key task of manufacturing IT on the way of organization towards |bold|cloud infrastructure|-|. ",
            "type": "imageWithText",
            "url": "https://res.cloudinary.com/oleksiikachan/image/upload/v1557089948/portfolio/images/projects/agw/agw_ip.png",
          },
          Object {
            "alignment": "right",
            "content": "|italic|AGW MES|-| is an |bold|ISA95|-| compliant |bold|web-based product|-| that provides a broad and flexible platform to optimize manufacturing operations. From simple to complex operations, |italic|AGW MES|-| manages the entire build process from start to finish providing operators detailed step-by-step directions and controlling their completion. |bold|Visualization|-| capabilities give you |bold|real-time visibility|-| and historical analytics over all manufacturing operations, enabling you to improve throughput, quality, agility and compliance. |italic|AGW MES|-| is a powerful, flexible and scalable |bold|Manufacturing Execution System|-| that is easily expanded with business requirements to maximize performance.",
            "type": "imageWithText",
            "url": "https://res.cloudinary.com/oleksiikachan/image/upload/v1557089948/portfolio/images/projects/agw/agw_mes.png",
          },
          Object {
            "alignment": "left",
            "content": "|italic|AGW WMS|-| provides control and visibility of your inventory from the time you receive a shipment until finished goods leave your warehouse. |italic|AGW WMS|-| is designed to |bold|optimize movement|-| of your inventory across your facility. From materials receiving down through the manufacturing process until packing and shipping to the customer |italic|AGW WMS|-| helps to direct your inventory in an optimal way providing |bold|up-to-the-minute information|-| on the quantity, location, status, and history of every item including work-in-process and finished goods.",
            "type": "imageWithText",
            "url": "https://res.cloudinary.com/oleksiikachan/image/upload/v1557089948/portfolio/images/projects/agw/agw_wms.png",
          },
          Object {
            "content": "The project work consisted of two parallel streams: development and support.",
            "type": "string",
          },
          Object {
            "header": "Development",
            "list": Array [
              "Implementing new features",
              "Overriding / Supporting legacy code (transfering one of the services from ASP.NET Forms to ASP.NET MVC)",
              "Implement different settings storage methods (as xml configuration, as database entity)",
              "Adjust database structure",
            ],
            "type": "list",
          },
          Object {
            "header": "Support",
            "list": Array [
              "Configuring & Deploying system on customers side",
              "Implement 'on demand' customer-specific features (batch loading, specific production routing)",
              "Production bug fixing",
            ],
            "type": "list",
          },
        ],
        "projectType": "development",
        "shortStack": ".NET Framework",
        "teamSize": "> 10 people",
        "title": "AGW. Automitive Manufacturing Solution",
      },
      Object {
        "client": Array [
          Object {
            "name": "Hyundai Motor Manufacturing Alabama",
            "url": "https://www.hmmausa.com/",
          },
        ],
        "coverImage": "https://picsum.photos/4302/2568",
        "externalResources": Array [
          Object {
            "name": "Product Page",
            "url": "http://www.catalystsi.com/index.php?option=com_content&view=article&id=8&Itemid=12",
          },
          Object {
            "name": "Company Page. OLSOM LLC",
            "url": "http://olsom.net/",
          },
          Object {
            "name": "Company Page. Catalyst Solutions Inc",
            "url": "http://www.catalystsi.com/",
          },
        ],
        "id": "tdm",
        "longStack": Array [
          ".NET Framework",
          "C#",
          "ASP.NET MVC",
          "ASP.NET Web Forms",
          "Javascript",
          "jQuery",
          "MS SQL Server",
          "Windows Service",
        ],
        "partner": Array [
          Object {
            "name": "OLSOM LLC",
            "url": "http://olsom.net/",
          },
          Object {
            "name": "Catalyst Solutions Inc",
            "url": "http://www.catalystsi.com/",
          },
        ],
        "projectDescription": Array [
          Object {
            "content": "|italic|Tools Data Manager (\\"TDM\\")|-| is an advanced manufacturing data collection and management solution that features comprehensive functionality for |bold|torque data capture|-| and management. |italic|TDM|-| is designed to simplify the challenge of connecting, extracting, associating, archiving, analyzing and mobilizing manufacturing data generated by localized shop-floor devices; such as, torque assembly systems, process PLCs, testing systems such as Labview, quality systems and etc. From |bold|quality & test data|-| to critical process data, |italic|TDM|-| aggregates all disparate localized manufacturing data into a single, central database and operating tool offering total control and collaboration.",
            "type": "string",
          },
          Object {
            "content": "|italic|TDM|-| is one of the first commercially designed software tools that integrates the function of |bold|device connectivity|-| into the core software tool including connectivity, data selection, data identity and many other useful functions.",
            "type": "string",
          },
          Object {
            "content": "The project work consisted of two parallel streams: development and support.",
            "type": "string",
          },
          Object {
            "header": "Development",
            "list": Array [
              "Implementing new features",
              "Adjust database structure",
            ],
            "type": "list",
          },
          Object {
            "header": "Support",
            "list": Array [
              "Configuring & Deploying system on customers side",
              "Implement 'on demand' customer-specific features (implement agw-like routing)",
              "Production bug fixing",
            ],
            "type": "list",
          },
        ],
        "projectType": "development",
        "shortStack": ".NET Framework",
        "teamSize": "> 10 people",
        "title": "TDM (Tools Data Manager). Manufacturing data collection and management solution",
      },
      Object {
        "coverImage": "https://res.cloudinary.com/oleksiikachan/image/upload/v1557369266/portfolio/images/projects/rpod/card.jpg",
        "externalResources": Array [
          Object {
            "name": "Official web site",
            "url": "https://rloop.org",
          },
          Object {
            "name": "Telemetry Station low-fidelity mock-up",
            "url": "https://res.cloudinary.com/oleksiikachan/image/upload/v1557356132/portfolio/images/projects/rpod/telemetry_mock.png",
          },
          Object {
            "name": "Telemetry Station final UI",
            "url": "https://res.cloudinary.com/oleksiikachan/image/upload/v1557356129/portfolio/images/projects/rpod/telemetry_ui.jpg",
          },
        ],
        "id": "rpod",
        "images": Array [
          Object {
            "imageUrl": "https://res.cloudinary.com/oleksiikachan/image/upload/v1557369266/portfolio/images/projects/rpod/cover.jpg",
          },
        ],
        "longStack": Array [
          "Adobe Illustrator",
          "Adobe Photoshop",
        ],
        "partner": Array [
          Object {
            "name": "rLoop Incorporated",
            "url": "https://www.rloop.org/",
          },
        ],
        "projectDescription": Array [
          Object {
            "content": "|bold|Hyperloop|-| is a conceptual, high-speed transportation system initially proposed by |bold|Elon Musk|-|. The concept consists of passenger and cargo pods being propelled at up to 760 mph in a low pressure tube using sustainable and cost-efficient energy. To accelerate its development, |bold|SpaceX|-| is hosted a competition for engineering teams to |bold|design|-| their own hyperloop pods and test them at their facility in |italic|Hawthorne, California|-|.",
            "type": "string",
          },
          Object {
            "content": "rLoop is |bold|open-sourced|-|, |bold|crowd-sourced|-| & |bold|crowsfunded|-| |italic|community|-|. I was a member of 2 teams: |bold|Systems Team|-| & |bold|Software Team|-| ",
            "type": "string",
          },
          Object {
            "header": "Software Team Member",
            "list": Array [
              "UX Research",
              "Telemetry Station UI Design",
              "Control Station UI Design",
              "Network Nodes development",
              "Leading small team < 5 people of designers & developers on UI/UX",
            ],
            "type": "list",
          },
          Object {
            "header": "Systems Team Member",
            "list": Array [
              "System Documentation creation",
              "Writing Human-Machine Interface section for Telemetry and Control stations in System Design Document (SDD)",
            ],
            "type": "list",
          },
          Object {
            "content": "I researched our processes and defined the sets of trackable data. Based on that I developed low-fidelity mock-ups.",
            "type": "string",
          },
          Object {
            "images": Array [
              Object {
                "alt": "Telemetry Station low-fidelity mock-up",
                "url": "https://res.cloudinary.com/oleksiikachan/image/upload/v1557366977/portfolio/images/projects/rpod/telemetry_mock.svg",
              },
              Object {
                "alt": "Telemetry Station final UI",
                "url": "https://res.cloudinary.com/oleksiikachan/image/upload/v1557356129/portfolio/images/projects/rpod/telemetry_ui.jpg",
              },
            ],
            "type": "images",
          },
        ],
        "projectType": "design",
        "shortStack": "UI & UX Design & Development",
        "teamSize": "> 20 people",
        "title": "Hyperloop rPod",
      },
      Object {
        "coverImage": "https://picsum.photos/4312/2869",
        "id": "cardboard-prod",
        "longStack": Array [
          "Ladder Logic",
          "Microsoft Visio",
        ],
        "partner": Array [
          Object {
            "name": "OLSOM LLC",
            "url": "http://olsom.net/",
          },
          Object {
            "name": "Agile Automation Technology LLC  ",
            "url": "http://www.agileautomation.net/",
          },
        ],
        "projectDescription": Array [
          Object {
            "content": "|bold|Reverse engineering|-| of |italic|ladder-logic|-| based corrugated industrial cardboard |bold|production software|-| to get complete networking system. System was developed using ladder logic so I had to understand what is going on there and then create some |bold|technical documentation|-| on how this system works.",
            "type": "string",
          },
        ],
        "projectType": "development",
        "shortStack": "Ladder Logic",
        "teamSize": "2 people",
        "title": "Corrugated Cardboard Production System",
      },
      Object {
        "coverImage": "https://picsum.photos/4312/2668",
        "id": "ftp-serivce",
        "longStack": Array [
          ".NET Framework",
          "C#",
          "Windows Service",
          "Windows OS",
          "Linux OS",
        ],
        "partner": Array [
          Object {
            "name": "OLSOM LLC",
            "url": "http://olsom.net/",
          },
        ],
        "projectDescription": Array [
          Object {
            "content": "|bold|Multi-threaded|-| Windows |italic|service|-| that transfers files locally and/or remotely via |bold|FTP|-| and is configured with *.ini file. All the service events, faults, warnings are logged into local txt file and |bold|Windows system log|-|.",
            "type": "string",
          },
        ],
        "projectType": "development",
        "shortStack": ".NET Framework",
        "teamSize": "2 people",
        "title": "FTP File Transfer Service",
      },
      Object {
        "client": Array [
          Object {
            "name": "Oleksii Kachan",
          },
        ],
        "coverImage": "https://res.cloudinary.com/oleksiikachan/image/upload/v1557512481/portfolio/images/projects/branding-oleksii-kachan/logo_square_light.png",
        "id": "branding-oleksii-kachan",
        "images": Array [
          Object {
            "imageUrl": "https://res.cloudinary.com/oleksiikachan/image/upload/v1557511177/portfolio/images/projects/branding-oleksii-kachan/details.png",
          },
        ],
        "longStack": Array [
          "Adobe Illustrator",
          "Adobe Photoshop",
        ],
        "projectDescription": Array [
          Object {
            "content": "My very first branding project was creating brand of myself. I defined 2 main colors, created logo, designed 2-side business card. Business card does not have title of person, because I do not want to put myself into frames of who I can be.",
            "type": "string",
          },
        ],
        "projectType": "design",
        "shortStack": "Personal Branding",
        "title": "Logo & Card Design",
      },
      Object {
        "client": Array [
          Object {
            "name": "Iryna Kachan",
          },
        ],
        "coverImage": "https://picsum.photos/4313/2868",
        "id": "branding-iryna-kachan",
        "longStack": Array [
          "Adobe Illustrator",
          "Adobe Photoshop",
        ],
        "projectDescription": Array [
          Object {
            "content": "I created brand for my mother as for financial services provider. I defined 2 main colors, created logo, designed 2-side business card.",
            "type": "string",
          },
        ],
        "projectType": "design",
        "shortStack": "Personal Branding",
        "title": "Logo & Card Design",
      },
      Object {
        "client": Array [
          Object {
            "name": "Oleksandr Gorchakov",
          },
        ],
        "coverImage": "https://picsum.photos/4312/2860",
        "id": "branding-oleksandr-grochakov",
        "longStack": Array [
          "Adobe Illustrator",
          "Adobe Photoshop",
        ],
        "projectDescription": Array [
          Object {
            "content": "I created brand for financial consultant from Ukraine. I defined 2 main colors, created logo, designed 2-side business card and created graphic candle-chart as visual element for business card. Candle-chart displays that person is connected with advanced financial operations.",
            "type": "string",
          },
        ],
        "projectType": "design",
        "shortStack": "Personal Branding",
        "title": "Logo & Card Design",
      },
    ]
  }
/>
`;
